genesyscloud/external_contacts_organization/resource_genesyscloud_externalcontacts_organization.go

```diff
@@ -0,0 +1,159 @@
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		externalOrganization, response, getErr := proxy.getExternalContactsOrganizationById(ctx, d.Id())
+		if getErr != nil {
+			if util.IsStatus404ByInt(response.StatusCode) {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read organization contact %s | error: %s", d.Id(), getErr), response))
+			}
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read organization contact %s | error: %s", d.Id(), getErr), response))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceExternalContactsOrganization(), 0, "")
+
+		resourcedata.SetNillableValue(d, "name", externalOrganization.Name)
+		resourcedata.SetNillableValue(d, "company_type", externalOrganization.CompanyType)
+		resourcedata.SetNillableValue(d, "industry", externalOrganization.Industry)
+		resourcedata.SetNillableValue(d, "primary_contact_id", externalOrganization.PrimaryContactId)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "address", externalOrganization.Address, flattenSdkAddress)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "phone_number", externalOrganization.PhoneNumber, flattenPhoneNumber)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "fax_number", externalOrganization.FaxNumber, flattenPhoneNumber)
+		resourcedata.SetNillableValue(d, "employee_count", externalOrganization.EmployeeCount)
+		resourcedata.SetNillableValue(d, "revenue", externalOrganization.Revenue)
+		resourcedata.SetNillableValue(d, "tags", externalOrganization.Tags)
+		resourcedata.SetNillableValue(d, "websites", externalOrganization.Websites)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "tickers", externalOrganization.Tickers, flattenTickers)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "twitter", externalOrganization.TwitterId, flattenSdkTwitterId)
+		resourcedata.SetNillableValue(d, "external_system_url", externalOrganization.ExternalSystemUrl)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "trustor", externalOrganization.Trustor, flattenTrustor)
+		resourcedata.SetNillableValueWithInterfaceArrayWithFunc(d, "external_data_sources", externalOrganization.ExternalDataSources, flattenExternalDataSources)
+		dataSchema, err := flattenDataSchema(externalOrganization.Schema)
+		if err != nil {
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("failed to flatten data schema %s | error: %s", d.Id(), getErr), response))
+		}
+		d.Set("schema", dataSchema)
+		cf, err := flattenCustomFields(externalOrganization.CustomFields)
+		if err != nil {
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("failed to flatten custom fields %s | error: %s", d.Id(), getErr), response))
+		}
```

Comment:
This response object is not related to the error. We should use `BuildDiagnosticError`. Also we should be returning `err` , not `getErr`. Same with the error check directly above